hatchling-web-app/src/components/TextEditor.jsx

```diff
@@ -57,6 +57,23 @@ export default function Texteditor() {
     }
   }
 
+  function onDownloadButtonClick() {
+    const fileName = prompt("Enter File Name");
```

Comment:
Good start. You will need to handle three use cases:  
1. They click cancel and no file should be saved
2. They enter a name without the proper extension; therefore, on should be append to the end of it
3. They enter a name with the proper extension; therefore, it should be saved as such

Reply:
4. They enter in an empty string or all whitespace. That should be considered canceled.